index.js

```diff
@@ -96,8 +96,7 @@ FibonacciHeap.prototype.extractMinimum = function () {
 
     // Merge the children of the minimum node with the root list
     this.minNode = mergeLists(nextInRootList, extractedMin.child, this.compare);
-    if (nextInRootList) {
-      this.minNode = nextInRootList;
+    if (this.minNode) {
```

Comment:
But this only happens when `extractedMin.next === extractedMin`, doesn't that mean it's the only tree in the root list? You only need to consolidate when there are 2+ trees in the root list.